src/realm/impl/transact_log.hpp

```diff
@@ -81,6 +81,8 @@ enum Instruction {
     instr_LinkListNullify = 36, // Remove an entry from a link list due to linked row being erased
     instr_LinkListClear = 37,   // Ramove all entries from a link list
     instr_LinkListSetAll = 38,  // Assign to link list entry
+    instr_AddSubtableSearchIndex = 39,  // Add search index to subtable
+    instr_RemoveSubtableSearchIndex = 40   // Remove search index from subtable
```

Comment:
Huh yeah there is definitely an API inconsistency here, nice observation.

However, I don't see how that must necessarily also impact the instruction? We should definitely be able to provide whatever API we prefer and keep it as one instruction (and then just hide any potential weirdness in the `TransactLogApplier`).